tests/unit/test_interval_regex.py

```diff
@@ -0,0 +1,34 @@
+import re
+import pytest
+
+cron_regex = re.compile(
+    r"^(?:-1|\d+(?:\.\d+)?|(((\d+,)+\d+|(\d+[/-]\d+)|\d+|\*(\/\d*)?) ?){5})$"
+)
+
+
+@pytest.mark.parametrize(
+    "is_valid, expression",
+    (
+        (True, "* * * * *"),
+        (True, "1 2 3 4 5"),
+        (True, "5 0 * 8 *"),
+        (True, "0 12 * * 3"),
+        (True, "0 22 * * 1-5"),
+        (True, "* 0,2 * * *"),
+        (True, "*/15 * * * *"),
+        (True, "*/15 0 1,15 * 1-5"),
+        (True, "0 0,12 1 */2 *"),
+        # (True, "* 0-20/2 * * *"), # 0-20/2 cron syntax is not supported by this regex...
+        (True, "1"),
+        (True, "-1"),
+        (True, "0"),
+        (True, "6371"),
+        (False, "-2"),
+        (False, "5 0 * 8 * 1"),
+        (False, "5 4 * * sun"),  # This syntax (sun) is not supported
+        (False, "@weekly"),  # This syntax is not supported
```

Comment:
Can you also check for the below cron schedules:

> | cron             | Expected | Actual |
> | * 0/0 * * *      | fail     | pass   |
> | * 0/9 * * *      | pass     | pass   |
> | * */9,9 * * *    | pass     | fail   |
> | * */9,9-12 * * * | pass     | fail   |
> | * 0/9,8 * * *    | pass     | fail   |
> | * 9/1,9-12 * * * | pass     | fail   |

Reply:
Here is the problem.
Writing the regex which will handle all these cases is a nightmare. 

Regarding your test cases,
1. - I'm not afraid of, splunk also handles correctness of CRON internally, so even if we fail here, we can rely on splunkd
3,4,5,6 - there are quite advanced CRON schedules, I'm not sure if supporting them is necessary (3 of them are even "non-standard).
But I will try to do smth with it

Reply:
hmm it looks like splunk supports only standard CRON format
```
* To specify a cron schedule, use the following format:
  * "<minute> <hour> <day of month> <month> <day of week>"
  * Cron special characters are acceptable. You can use combinations of "*",
  ",", "/", and "-" to specify wildcards, separate values, specify ranges
  of values, and step values.
* The cron implementation for data inputs does not currently support names
  of months or days.
```

Reply:
There is a new regex which handles all of these cases, but not the 1st one - but I guess such syntax is just an user error, so that we don't have to handle it. It would be much more complicated